Add support for common @-tags

abgruszecki opened the pull request.

abgruszecki committed:
Add support for common "at-sections" of Scaladoc

This includes:
- resolving errors like Scaladoc does
- displaying repeated at-sections in a reasonable order
- ensuring that the sections in Markdown get parsed properly (see @example)

abgruszecki committed:
Add CSS for params doc

abgruszecki committed:
Nest appropriate at-sections

pikinier20 commented on src/main/scala/dotty/dokka/tasty/ScalaDocSupport.scala:
```diff
@@ -48,13 +49,39 @@ trait ScaladocSupport { self: TastyParser =>
     addOpt(parsed.since)(dkkd.Since(_))
+    addOpt(parsed.deprecated)(dkkd.Deprecated(_))
+    addSeq(parsed.todo)(dkkd.CustomTagWrapper(_, "Todo"))
+    addSeq(parsed.see)(dkkd.CustomTagWrapper(_, "See"))
```
In dokka there is special TagWrapper for `See`

abgruszecki replied:
Right, that's a good catch. I'll actually need to do more work here - all of these tags should be repeatable, but currently all NamedTags with the same name are deduplicated. This is also why I didn't use the original Dokka classes - they are all named, but deduplicating them is not at all what we want.

pikinier20 commented on src/main/scala/dotty/dokka/model/scalaModel.scala:
```diff
@@ -61,3 +62,37 @@ case class HtmlContentNode(
+      tag.getChildren.get(1),
+    )
+  }
+}
```
There's hack that can be used to extend sealed classes from Kotlin in Scala: you need to pass `null` as parameter to constructor of sealed class e. g.:
We have Kotlin class like that:
```Kotlin
sealed class ExtraModifiers(val name: String)
```
We can extend it by doing:
```Scala
enum ScalaOnlyModifiers(val name: String, val prefix: Boolean) extends ExtraModifiers(name, null)
```
We already do this trick few times and it might be nicer to extend `NamedTagWrapper` than hack fields

abgruszecki replied:
Right, I forgot that. I'll fix this code then.

abgruszecki committed:
Add an explanatory comment to potentially unsafe code

abgruszecki committed:
Define custom TagWrappers by inheriting from sealed Kotlin class

Define special "sequential" tags, as well as refactor HackNestedTagWrapper to be
a normal class instead.

romanowski commented on src/main/scala/dotty/dokka/tasty/comments/Comments.scala:
```diff
@@ -49,9 +52,10 @@ case class PreparsedComment (
-  groupDesc:               Map[String, String],
-  groupNames:              Map[String, String],
-  groupPrio:               Map[String, String],
+  // NOTE these don't need to be sorted in principle, but code is nicer if they are
```
good call!
